python-job-crawler.py

```diff
@@ -1,88 +1,143 @@
 
-def close_dialogs(driver):
+
+def close_dialogs(driver: webdriver.Chrome) -> None:
+    """
+    Tries to close any potential pop-up dialogs that may appear on the page.
+
+    Args:
+        driver (webdriver.Chrome): The Selenium WebDriver instance used for browsing.
+
+    Returns:
+        None
+    """
     try:
-        # If there's a dialog box with a close button with class .close-button or .close
-        # Add more class names as you encounter them
-        close_button = driver.find_element(By.CSS_SELECTOR, '.close-button' | '.close')
+        # Modify the CSS selector to find common close buttons, adjust as needed
+        close_button = WebDriverWait(driver, PAGE_LOADING_TIME_SLEEPER).until(
+            EC.element_to_be_clickable((By.CSS_SELECTOR, '.close-button, .close'))
+        )
         if close_button:
             close_button.click()
+            logging.info("Closed a dialog on the page.")
+    except TimeoutException:
+        logging.debug("No dialog found to close.")
     except Exception as e:
-        # No dialog to close or unable to find the close button
-        pass
+        logging.error(f"An unexpected error occurred: {e}")
 
-def crawl_careerPage(url, driver):
-    driver.get(url)
-    time.sleep(5)  # Wait for initial load
 
-    # Handle any potential dialogs
-    close_dialogs(driver)
+def crawl_careerPage(url: str, driver: webdriver.Chrome, kws: List[str]) -> bool:
+    """
+    Crawls a given career page URL to search for keywords in the job listings.
+
+    Args:
+        url (str): URL of the career page to visit.
+        driver (webdriver.Chrome): Selenium WebDriver instance used for browsing.
+        kws (List[str]): the kws to look for on page at URL
+
+    Returns:
+        bool: True if any keywords were found on the page, False otherwise.
+    """
+    logging.info(f"Visiting career page: {url}")
+
+    driver.get(url)  # Selenium is loading url page
 
+    # Wait for the page to load completely using WebDriverWait before moving on
     try:
-        # Wait for page to fully load
-        WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.TAG_NAME, "body")))
+        WebDriverWait(driver, PAGE_LOADING_TIME_SLEEPER).until(
+            EC.presence_of_element_located((By.TAG_NAME, "body"))
+        )
     except Exception as error:
-        print(f"Error loading {url}: {error}")
+        logging.error(f"Error loading {url}: {error}")
         return False
 
+    # Handle any potential dialogs after the page loads
+    close_dialogs(driver)
+
+    # Parse the page with BeautifulSoup
     soup = BeautifulSoup(driver.page_source, 'html.parser')
-    # Convert page text to lowercase to make search case-agnostic
     page_text = soup.get_text().lower()
-    found_keywords = set()
-    for keyword in keywords:
-        if keyword.lower() in page_text:
-            found_keywords.add(keyword)
+
```

Comment:
Above line 130 page_text = soup.get_text().lower() please insert the comment back in:
`# Convert page text to lowercase to make search case-agnostic`